skops/cli/_update.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+import argparse
+import json
+import logging
+import zipfile
+from pathlib import Path
+
+from skops.cli._utils import get_log_level
+from skops.io import dump, load
+from skops.io._protocol import PROTOCOL
+
+
+def _update_file(
+    input_file: str | Path,
+    output_file: str | Path | None = None,
+    inplace: bool = False,
+    logger: logging.Logger = logging.getLogger(),
+) -> None:
+    """Function that is called by ``skops update`` entrypoint.
+
+    Loads a skops model from the input path, converts to the current skops format, and
+    saves to output file.
+
+    Parameters
+    ----------
+    input_file : str, or Path
+        Path of input skops model to load.
+
+    output_file : str, or Path, default=None
+        Path to save the updated skops model to.
+
+    inplace : bool, default=False
+        Whether to update and overwrite the input file in place.
+
+    logger : logging.Logger, default=logging.getLogger()
+        Logger to use for logging.
+    """
+    if inplace:
+        if output_file is None:
+            output_file = input_file
+        else:
+            raise ValueError(
+                "Cannot specify both an output file path and the inplace flag. Please"
+                " choose whether you want to create a new file or overwrite the input"
+                " file."
+            )
+
+    input_model = load(input_file, trusted=True)
+    with zipfile.ZipFile(input_file, "r") as zip_file:
+        input_file_schema = json.loads(zip_file.read("schema.json"))
+
+    if input_file_schema["protocol"] != PROTOCOL:
```

Comment:
I'm just wondering about the case of the input file having a _higher_ protocol than the current one. Might be a niche application, but could have it's uses. The code would already work as is, but the message might be confusing. I wonder: Do we want to change the wording to "upgrade" / "downgrade" depending on whether the protocol was higher or lower?

Reply:
hehe, I think it's fair to not support downgrade at least for now?

Reply:
Hmm, not sure, I could imagine edge cases (like if we mess up an upgrade of the protocol) where users might want to downgrade. Probably rare (we never mess up), but should it be impossible?

Reply:
I just mean to keep it simple to start with, and add that usecase if we need to.

Reply:
I'd be okay with either solution, but it should be clear to the user what's happening. Treating upgrade and downgrade identically is bound to create confusion.

Reply:
I have added a if-statement to check for a _higher_ protocol and exit with a message if true. WDYT?